typescript/cosmos-sdk/src/clients/protocol.ts

```diff
@@ -0,0 +1,44 @@
+import {
+  AltVM,
+  ChainMetadataForAltVM,
+  ITransactionSubmitter,
+  ProtocolProvider,
+  SignerConfig,
+  TransactionSubmitterConfig,
+} from '@hyperlane-xyz/provider-sdk';
+import { IProvider } from '@hyperlane-xyz/provider-sdk/altvm';
+import { AnnotatedTx, TxReceipt } from '@hyperlane-xyz/provider-sdk/module';
+import { assert } from '@hyperlane-xyz/utils';
+
+import { CosmosNativeProvider } from './provider.js';
+import { CosmosNativeSigner } from './signer.js';
+
+export class CosmosNativeProtocolProvider implements ProtocolProvider {
+  createProvider(chainMetadata: ChainMetadataForAltVM): Promise<IProvider> {
+    assert(chainMetadata.rpcUrls, 'rpc urls undefined');
+    const rpcUrls = chainMetadata.rpcUrls.map((rpc) => rpc.http);
+    return CosmosNativeProvider.connect(rpcUrls, chainMetadata.domainId);
+  }
+
+  async createSigner(
+    chainMetadata: ChainMetadataForAltVM,
+    config: SignerConfig,
+  ): Promise<AltVM.ISigner<AnnotatedTx, TxReceipt>> {
+    assert(chainMetadata.rpcUrls, 'rpc urls undefined');
+    const rpcUrls = chainMetadata.rpcUrls.map((rpc) => rpc.http);
+
+    const { privateKey, ...extraParams } = config;
+    return CosmosNativeSigner.connectWithSigner(
+      rpcUrls,
+      privateKey,
+      extraParams,
+    );
+  }
+
+  createSubmitter<TConfig extends TransactionSubmitterConfig>(
+    _chainMetadata: ChainMetadataForAltVM,
+    _config: TConfig,
+  ): Promise<ITransactionSubmitter> {
+    throw Error('Not implemented');
```

Comment:
Provably worth adding a todo here specifying that this will be implemented in a follow up PR

Reply:
added